src/util/colorScale.js

```diff
@@ -120,10 +120,20 @@ export const calcColorScale = (colorBy, controls, tree, treeToo, metadata) => {
     console.error("calcColorScale called before tree is ready.");
     error = true;
   } else if (genotype) { /* G E N O T Y P E */
-    legendValues = orderOfGenotypeAppearance(tree.nodes);
+    legendValues = orderOfGenotypeAppearance(tree.nodes, controls.mutType);
+    const trueValues = controls.mutType === "nuc" ? legendValues.filter((x) => x !== "X" && x !== "-" && x !== "N") :
+      legendValues.filter((x) => x !== "X" && x !== "-");
+    const unkValues = controls.mutType === "nuc" ? legendValues.filter((x) => x === "X" || x === "-" || x === "N") :
+      legendValues.filter((x) => x === "X" || x === "-");
+    let domain = [undefined, ...trueValues];
+    let range = [unknownColor, ...genotypeColors.slice(0, trueValues.length)];
+    if (unkValues.length) { // we must add these to the domain + provide a value in the range
+      domain = domain.concat(unkValues);
+      range = range.concat(createListOfColors(unkValues.length, [rgb(192, 192, 192), rgb(32, 32, 32)]));
```

Comment:
Instead of inferring colours between 2 values why don't we define 3 shades of grey for the three potential unknown "bases"? That way the colours for `N`, `X` and `-` will be the same over different datasets